### ISSUE TITLE
Feature request: Support https as upstream protocol

### ISSUE THREAD

**Issue by sgofferj** [JrCs/docker-letsencrypt-nginx-proxy-companion#578]
I'm currently running into issue with Collabora/Libre Office Online. The Collabora container defaults to running SSL and creates its own self-signed certs. The latter can be turned off and with clever bind volumes the certs from your container could be used.
I'm reasonably sure that there's other apps/services which also run SSL per default.
In the case of Collabora SSL can even be switched off completely, in which case Nextcloud starts whining about protocol mismatches despite Collabora being behind the reverse proxy.

Long story short, an optional ENV var which allows to select SSL as upstream protocol would be quite useful, because it would allow running everything through one proxy, making full use of SNI and not drilling many holes in the firewall.

**Comment by buchdag** [comment]
Hi.

Short answer : https://github.com/jwilder/nginx-proxy#ssl-backends

This project (`JrCs/docker-letsencrypt-nginx-proxy-companion`) "only" purpose is to:

-  enable the automatic creation of certificates from containers env var.
-  put them in the right place with the right name so they can be used by nginx-proxy docker-gen template.
-  reload the nginx proxy when needed (cert creation or renewal).

What you want to achieve depends entirely on the [`jwilder/nginx-proxy`](https://github.com/jwilder/nginx-proxy) project, which does the actual proxying and is maintained by different peoples.

We only deal with issues related to letsencrypt-nginx-proxy-companion, so [please submit your nginx-proxy issues directly to the nginx-proxy repo.](https://github.com/jwilder/nginx-proxy/issues)
